karabo_data/validation.py

```diff
@@ -30,8 +31,36 @@ def validate(self):
     def run_checks(self):
         self.problems = []
         self.check_indices()
+        self.check_trainids()
         return self.problems
 
+    def record(self, msg, **kwargs):
+        self.problems.append(dict(
+            msg=msg, file=self.filename, **kwargs
+        ))
+
+    def check_trainids(self):
+        ds_path = 'INDEX/trainId'
+        train_ids = self.file.file[ds_path][:]
+
+        if (train_ids == 0).any():
+            first0 = train_ids.tolist().index(0)
+            if not (train_ids[first0:] == 0).all():
+                self.record(
+                    'Zeroes in trainId index before last train ID',
+                    dataset=ds_path,
+                )
+            nonzero_tids = train_ids[train_ids != 0]
+        else:
+            nonzero_tids = train_ids
+
+        if len(nonzero_tids) > 1:
+            if not (nonzero_tids[1:] > nonzero_tids[:-1]).all():
```

Comment:
I would assume that they need to increase monotonically (by 1) but we better check with IT.